src/context/auth-context.js

```diff
@@ -14,24 +15,31 @@ const AuthProvider = ({ children }) => {
       response: loginUserData,
       error: loginError,
       apiCall: setLoginResponse,
+      setError: resetLoginError,
    } = useAxios();
 
    const {
       response: signupUserData,
       error: signupError,
       apiCall: setSignupResponse,
+      setError: resetSignupError,
    } = useAxios();
 
    //if token is present in localstorage then show user logged in
    useEffect(() => {
       if (localStorage.getItem("token") !== null) {
-         authDispatch({ type: "TOGGLE_LOGIN", payload: true });
+         authDispatch({
+            type: "SET_USER_CREDENTIALS",
+            payload: getLocalData("user"),
+         });
       }
+      // eslint-disable-next-line
    }, []);
```

Comment:
This needs a change of logic!
You can't just say a user exists by checking if the token is null.
A token in local storage can be tampered with meaning someone can edit the token directly!
You need to verify the token to see if the user exists or not by making a call to the backend

Reply:
Thank you very much Sudip, for pointing this out. I have fixed this issue.